cmd/ci-operator/main_test.go

```diff
@@ -543,3 +543,119 @@ func TestExcludeContextCancelledErrors(t *testing.T) {
 		}
 	}
 }
+
+func TestMultiStageParams(t *testing.T) {
+	testCases := []struct {
+		id             string
+		inputParams    stringSlice
+		expectedParams map[string]string
+		testConfig     []api.TestStepConfiguration
+		expectedErrs   []string
+	}{
+		{
+			id:          "MultiStageTestConfigurationLiteral with valid params",
+			inputParams: stringSlice{[]string{"PARAM1=VAL1", "PARAM2=VAL2"}},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfigurationLiteral: &api.MultiStageTestConfigurationLiteral{
+						Environment: map[string]string{
+							"OTHERPARAM": "OTHERVAL",
+						},
+					},
+				},
+			},
+			expectedParams: map[string]string{
+				"PARAM1":     "VAL1",
+				"PARAM2":     "VAL2",
+				"OTHERPARAM": "OTHERVAL",
+			},
+		},
+		{
+			id:          "MultiStageTestConfiguration with valid params",
+			inputParams: stringSlice{[]string{"PARAM1=VAL1", "PARAM2=VAL2"}},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfiguration: &api.MultiStageTestConfiguration{
+						Environment: map[string]string{
+							"OTHERPARAM": "OTHERVAL",
+						},
+					},
+				},
+			},
+			expectedParams: map[string]string{
+				"PARAM1":     "VAL1",
+				"PARAM2":     "VAL2",
+				"OTHERPARAM": "OTHERVAL",
+			},
+		},
+		{
+			id:          "Override test param",
+			inputParams: stringSlice{[]string{"PARAM1=NEWVAL", "PARAM2=VAL2"}},
+			testConfig: []api.TestStepConfiguration{
+				{
+					MultiStageTestConfigurationLiteral: &api.MultiStageTestConfigurationLiteral{
+						Environment: map[string]string{
+							"PARAM1": "VAL1",
+						},
+					},
+				},
+			},
+			expectedParams: map[string]string{
+				"PARAM1": "NEWVAL",
+				"PARAM2": "VAL2",
+			},
+		},
+		{
+			id:             "invalid params",
+			inputParams:    stringSlice{[]string{"PARAM1", "PARAM2"}},
+			expectedParams: map[string]string{},
+			expectedErrs: []string{
+				"could not parse multi-stage-param: PARAM1 is not in the format key=value",
+				"could not parse multi-stage-param: PARAM2 is not in the format key=value",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
```

Comment:
Check out how other tests use subtests with `t.Run` and `t.Parallel` (https://golang.org/pkg/testing/) to improve the experience for whoever will run this.